### ISSUE TITLE
Can't type narrow a discriminated union instance variable

### ISSUE THREAD

**Issue by trygveaa** [microsoft/pyright#4548]
**Describe the bug**

If you have a instance variable which is a discriminated union type, it's not possible to type narrow it in a function.

**To Reproduce**

Create a class with an instance variable which has a discriminated union type. Try to type narrow it by checking it in a function. pyright will fail to narrow it unless the instance variable is assigned to a local variable and that local variable is used.

An example is provided below. Running this gives:

```
test.py:29:25 - information: Type of "self.response" is "Success | Error"
test.py:30:13 - error: Could not access item in TypedDict
  "result" is not a defined key in "Error" (reportGeneralTypeIssues)
test.py:35:25 - information: Type of "response" is "Success"
```

**Expected behavior**

pyright should be able to type narrow the instance variable without needing to use a local variable.

**Screenshots or Code**

```python
from typing import Literal, TypedDict, final


@final
class Success(TypedDict):
    ok: Literal[True]
    result: str


@final
class Error(TypedDict):
    ok: Literal[False]
    error: str


Response = Success | Error


def get_response() -> Response:
    ...


class Test:
    def __init__(self):
        self.response: Response = get_response()

    def fails(self):
        if self.response["ok"] is True:
            reveal_type(self.response)
            self.response["result"]

    def working(self):
        response = self.response
        if response["ok"] is True:
            reveal_type(response)
            response["result"]
```

**VS Code extension or command-line**

Running version 1.1.291 from the command-line.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.292, which I just published. It will also be included in a future release of pylance.
